spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -526,8 +540,7 @@ else if (payload instanceof byte[] || payload instanceof String) {
 			else if (payload instanceof InputStream) {
 				return new StreamHolder((InputStream) payload, "InputStream payload");
 			}
-			else if (payload instanceof Resource) {
-				Resource resource = (Resource) payload;
+			else if (payload instanceof Resource resource) {
```

Comment:
In future, please refrain from this kind of (unrelated) refactoring when back porting is required (I will fix this time during back port).

Reply:
Doh; sorry - I didn't notice that we stop at 6.0.x.